src/python/WMComponent/AnalyticsDataCollector/AnalyticsPoller.py

```diff
@@ -142,10 +144,18 @@ def algorithm(self, parameters):
 
             self.localSummaryCouchDB.uploadData(requestDocs)
             logging.info("Request data upload success\n %s request, \nsleep for next cycle" % len(requestDocs))
-            DataUploadTime.setInfo(uploadTime, "ok")
+
+            self.centralWMStatsCouchDB.updateAgentInfoInPlace(self.agentInfo["agent_url"],
+                                                              {"data_last_update": uploadTime, "data_error": "ok"})
 
         except Exception as ex:
             logging.error("Error occurred, will retry later:")
             logging.error(str(ex))
-            DataUploadTime.setInfo(False, str(ex))
             logging.error("Trace back: \n%s" % traceback.format_exc())
+
+            try:
+                self.centralWMStatsCouchDB.updateAgentInfoInPlace(self.agentInfo["agent_url"], {"data_error": str(ex)})
```

Comment:
Just trying to understand. If there was an exception above, we immediately post this error to couch wmstats, which is visible through the agent_info view? Does it show up like a component error or something else?

Reply:
Yes it will be visible in agent_info view. It should show up "data_error".